test/features/calendar.feature

```diff
@@ -0,0 +1,167 @@
+            | description | "descriptive Description"             |
+            | url         | "https://domain.com"                  |
+            | platform    | "YouTube"                             |
+            | author      | {"platform":"discord","uid":"hubber"} |
+            | startDate   | "2021-01-01T00:00:00.000Z"            |
+            | endDate     | "2021-01-01T00:00:00.000Z"            |
+            | createdOn   | "2021-01-01T00:00:00.000Z"            |
+            | updatedOn   | "2021-01-01T00:00:00.000Z"            |
+
+    Scenario: create a event without authorization
+        Given make a POST request to "/calendar" with:
+            | name        | "Livestream XY"                       |
+            | description | "descriptive Description"             |
+            | url         | "https://domain.com"                  |
+            | platform    | "YouTube"                             |
+            | author      | {"platform":"discord","uid":"hubber"} |
+            | startDate   | "2021-01-01T00:00:00.000Z"            |
+            | endDate     | "2021-01-01T00:00:00.000Z"            |
+        Then the response status code should be 401
+        And the response should contains:
+            | statusCode | 401            |
+            | message    | "Unauthorized" |
+
+    Scenario: get sorted ongoing and future events
+        Given authorisation
+        And make a POST request to "/calendar" with:
+            | name        | "Livestream XY"                       |
+            | description | "descriptive Description"             |
+            | url         | "https://domain.com"                  |
+            | platform    | "YouTube"                             |
+            | author      | {"platform":"discord","uid":"hubber"} |
+            | startDate   | "2022-01-01T00:00:00.000Z"            |
+            | endDate     | "2023-01-01T00:00:00.000Z"            |
+        When make a POST request to "/calendar" with:
+            | name        | "Livestream YZ"                      |
+            | description | "undescriptive Description"          |
+            | url         | "https://mydomain.com"               |
+            | platform    | "Twitch"                             |
+            | author      | {"platform":"discord","uid":"hubby"} |
+            | startDate   | "2021-01-01T00:00:00.000Z"           |
+            | endDate     | "2022-01-01T00:00:00.000Z"           |
+        When make a GET request to "/calendar"
+        Then the response status code should be 200
+        And the response should contains:
+            | future  | [{"id":123,"name":"Livestream XY","description":"descriptive Description","url":"https://domain.com","platform":"YouTube","author":{"platform":"discord","uid":"hubber"},"startDate":"2022-01-01T00:00:00.000Z","endDate":"2023-01-01T00:00:00.000Z","createdOn":"2021-01-01T00:00:00.000Z","updatedOn":"2021-01-01T00:00:00.000Z"}]   |
```

Comment:
I think we need to create another step def similar to this one that would make it easier to read - I can do it in another PR after this one is merged.

```ts
@then(/the response in item "([^"]*)" should contains:/)
public dataResponseItemTable(item: number, table: { rawTable: [] }) {
    const data = this.context.tableToObject(table);
    expect(JSON.parse(this.context.response.text)[item]).to.to.eql(data);
}
```
https://github.com/EddieHubCommunity/api/blob/main/test/step-definitions/responses.ts#L31

The new step def would be something like this

```ts
@then(/the response property "([^"]*)" in field "([^"]*)" should contains:/)
public dataResponseItemTable(property: string, item: number, table: { rawTable: [] }) {
    const data = this.context.tableToObject(table);
    expect(JSON.parse(this.context.response.text)[property][item]).to.eql(data);
}
```

The tests would be something like

```gherkin
And  the response property "future" in item "0" should contains:
            | id   | 123                |
            | name | "Livestream XY"    |
And  the response property "ongoing" in item "0" should contains:
            | id   | 123                |
            | name | "Livestream YZ"    |
```

Note: I think `contains` should be singular 🤔 - this can be changed later